### DISCUSSION TITLE
JavaScript heap out of memory

### DISCUSSION THREAD

**Discussion by clauBv23** [NomicFoundation/hardhat#5943] [Troubleshooting]
Hi there I'm receiving 
```FATAL ERROR: Reached heap limit Allocation failed - JavaScript heap out of memory```
 when running `hh coverage`. I have seen that this issue is normally related to old versions of hardhat and/or solidity-coverage however I started having the error after upgrading `hardhat` to the latest. I might have other related repos outdated tho.

Is there any idea of what could be happening, how fix this problem, or what repos I could try to upgrade to fix this? 

This are my current dependencies 
```"@ethersproject/abi": "5.7.0",
    "@ethersproject/abstract-signer": "5.7.0",
    "@ethersproject/bignumber": "5.7.0",
    "@ethersproject/bytes": "5.7.0",
    "@ethersproject/constants": "5.7.0",
    "@ethersproject/providers": "5.7.2",
    "@nomicfoundation/hardhat-chai-matchers": "^1.0.6",
    "@nomicfoundation/hardhat-network-helpers": "^1.0.10",
    "@nomicfoundation/hardhat-toolbox": "^2.0.2",
    "@nomiclabs/hardhat-ethers": "^2.2.3",
    "@nomiclabs/hardhat-etherscan": "^3.1.7",
    "@openzeppelin/hardhat-upgrades": "^1.28.0",
    "@typechain/ethers-v5": "^10.1.1",
    "@typechain/hardhat": "^6.1.4",
    "@types/chai": "^4.3.4",
    "@types/mocha": "^10.0.0",
    "@types/node": "^18.11.9",
    "chai": "^4.3.7",
    "cross-env": "^7.0.3",
    "dotenv": "^16.3.1",
    "ethers": "5.7.2",
    "hardhat": "^2.22.15",  // previously was using v2.13.1
    "hardhat-deploy": "^0.11.26",
    "hardhat-gas-reporter": "^1.0.9",
    "mocha": "^10.1.0",
    "rimraf": "^5.0.5",
    "solidity-coverage": "^0.8.13",   // previously was using v0.8.5
    "solidity-docgen": "^0.6.0-beta.36",
    "tmp-promise": "^3.0.3",
    "ts-node": "^10.9.1",
    "typechain": "^8.3.2",
    "typescript": "5.2.2"
```

Increasing the size of node options solves the issue, but since contract size and test count are in normal ranges, and they used to work previously, I would like to identify the root cause of the problem.

EDIT: 
it works with `"solidity-coverage": "0.8.2"` and ` "hardhat": "2.12.7"`

**Comment by kanej** [comment]
Is it only for coverage you are seeing the out of memory?

**Reply by clauBv23** [reply]
Hey @kanej, yes only when running `hh coverage`, `hh test` works normally.

**Reply by kanej** [reply]
`solidity-coverage` has had memory issues. We suspect this is a combination of `solidity-coverage` with changes we have made in EDR.
`solidity-coverage` has docs on the out of memory issues: https://github.com/sc-forks/solidity-coverage/blob/master/docs/faq.md#running-out-of-memory
We are working on Hardhat 3 and we are bringing coverage into the core of Hardhat to reduce these sorts of issues.
